frontend/packages/operator-lifecycle-manager/src/components/descriptors/index.tsx

```diff
@@ -8,133 +8,202 @@ import { JSONSchema6 } from 'json-schema';
+  const description = descriptor?.description || groupSchema?.description;
+  const label = descriptor?.displayName || groupSchema?.title || _.startCase(groupPath);
+  const arrayGroups = _.pickBy(nested, 'isArrayGroup');
+  const primitives = _.omitBy(nested, 'isArrayGroup');
+  const className = _.isEmpty(arrayGroups) ? 'col-sm-6' : 'col-sm-12';
   return (
-    <dl className={`olm-descriptors olm-descriptors--${type}`}>
-      {_.map(groupedDescriptors, (group: Descriptor[], groupName) => (
-        <div key={`${type}.${groupName}`} className={itemClassName}>
-          {group.length > 1 ? (
-            <DescriptorDetailsItemGroup
-              descriptors={group}
-              groupName={groupName}
-              model={model}
-              obj={obj}
-              onError={onError}
-              schema={schema}
-              type={type}
-            />
-          ) : (
-            <DescriptorDetailsItem
-              descriptor={group[0]}
-              model={model}
-              obj={obj}
-              onError={onError}
-              schema={schema}
-              type={type}
-            />
-          )}
-        </div>
-      ))}
-    </dl>
+    <div className={className}>
+      <DetailsItem description={description} label={label} obj={obj} path={`${type}.${groupPath}`}>
+        <dl>
+          <div className="details-item__value--group">
+            <Flex justifyContent={{ default: 'justifyContentFlexStart' }}>
+              {!_.isEmpty(primitives) && (
+                <FlexItem grow={{ default: 'grow' }}>
+                  {_.map(primitives, ({ descriptor: primitiveDescriptor }) => (
+                    <DescriptorDetailsItem
+                      descriptor={primitiveDescriptor}
+                      key={`${type}.${primitiveDescriptor.path}`}
+                      model={model}
+                      obj={obj}
+                      onError={onError}
+                      schema={schema}
+                      type={type}
+                    />
+                  ))}
+                </FlexItem>
+              )}
+              {!_.isEmpty(arrayGroups) && (
+                <FlexItem grow={{ default: 'grow' }}>
+                  {_.map(arrayGroups, (arrayGroup: DescriptorGroup) => (
+                    <DescriptorDetailsItemArrayGroup
+                      group={arrayGroup}
+                      groupPath={arrayGroup.arrayGroupPath}
+                      key={`${type}.${groupPath}.${arrayGroup.arrayGroupPath}`}
+                      model={model}
+                      obj={obj}
+                      onError={onError}
+                      schema={schema}
+                      type={type}
+                    />
+                  ))}
+                </FlexItem>
+              )}
+            </Flex>
+          </div>
+        </dl>
+      </DetailsItem>
+    </div>
   );
 };
 
+export const DescriptorDetailsItemList = withFallback<DescriptorDetailsItemListProps>(
```

Comment:
This moves the fallback around all descriptors, correct? My concern is one bad descriptor would prevent all of the others from displaying.

Reply:
True, I can move it back down to the individual descriptor level.